vueui/src/components/view/devices/aicard.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <div>
+    <template v-if="has_ai">
+      <vui-group>
+        <vui-group-item label="Hardware Integrity">{{hardware_integrity}}%</vui-group-item>
+        <vui-group-item label="Backup Capacitor">{{backup_capacitor}}%</vui-group-item>
+      </vui-group>
+
+      <template v-if="has_laws">
+        <table class="borders">
+          <tr><td class="law_index">Index</td><td>Law</td></tr>
+            
+          <div class="itemLabelNarrow">
+            Laws:
+          </div>
+          <tr v-for="law in laws" :key="law.index">
+            <td valign="top">{{law.index}}</td>
+            <td>{{law.law}}</td>
+          </tr>
+        </table>
+      </template>
+      <span class="notice" v-else>
+        No laws found.
+      </span>
+
+      <vui-group v-if="operational">
+        <vui-group-item label="Radio Subspace Transceiver">
+          <vui-button :class="{selected: radio}" :params="{ radio: 0 }">Enabled</vui-button>
+          <vui-button :class="{danger: !radio}" :params="{ radio: 1 }">Disabled</vui-button>
+        </vui-group-item>
+
+        <vui-group-item label="Wireless Interface">
+          <vui-button :class="{selected: wireless}" :params="{ wireless: 0 }">Enabled</vui-button>
+          <vui-button :class="{danger: !wireless}" :params="{ wireless: 1 }">Disabled</vui-button>
+        </vui-group-item>
+
+        <vui-group-item v-if="flushing">
+          <span class="notice">AI wipe in progress...</span>
+        </vui-group-item>
+        <vui-group-item v-else label="Wipe AI">
+          <vui-button class="danger" :params="{ wipe: 1 }">Wipe</vui-button>
+        </vui-group-item>
+      </vui-group>
+    </template>
+    <template v-else>
+      Stored AI: <span class="notice">No AI detected.</span>
+    </template>
+  </div>
+</template>
+
+<script>
+export default {
+  data() {
+    return this.$root.$data.state;
+  }
+};
+</script>
+
+<style lang="scss" scoped>
+table.borders   {
+    width:95%; 
+    margin-left:2.4%; 
+    margin-right:2.4%;
+}
+
+table.borders, table.borders tr {
+    border: 1px solid White;
+}
+
+td.law_index {
+    width: 50px;
+}
+
+td.state {
+    width: 63px;
+    text-align:center; 
+}
+
+td.add {
+    width: 36px;
+}
+
+td.edit {
+    width: 36px;
+    text-align:center; 
+}
+
+td.delete {
+    width: 53px;
+    text-align:center; 
+}
+
+td.law_type {
+    width: 65px;
+}
+
+td.position {
+    width: 37px;
+}
```

Comment:
Try to remove styles and check if they are really needed. Also using proper SCSS nesting would be nice, because currently this is just CSS.

Reply:
Ye idno what scss is söö we're stuck with this. It works and isn't offensive either way.